alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/RestIndexDestinationAction.kt

```diff
@@ -117,6 +117,7 @@ class RestIndexDestinationAction(
             if (!scheduledJobIndices.scheduledJobIndexExists()) {
                 scheduledJobIndices.initScheduledJobIndex(ActionListener.wrap(::onCreateMappingsResponse, ::onFailure))
             } else {
+                scheduledJobIndices.updateScheduledJobIndex()
```

Comment:
While I am sure this solution works. Rather than performing this request on every indexing operation of monitors / destinations it would be great to check if the schema version has changed, if it hasn't there is no need to perform the request.

Reply:
Yes, agree. Actually, I have put the checking logic in "SchemaVersionUtils.shouldUpdateIndex", if the schema version not changed, will skip updating the index mapping.